### PR TITLE
Feature/pelagos 3359 add forgot username route and implementation

### PR DESCRIPTION
**Pull request by praneethpr**

### CODE REVIEW

File: src/Pelagos/Bundle/AppBundle/Resources/views/Email/Account/UsernameRetrieval.email.twig [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+{% endblock %}
+
+{% block body_text %}
+    Dear {{ person.getFirstName }} {{ person.getLastName }},
+
+    You have recently requested to retrieve your username using the Forgot Username form on our system.
+
+    username: person.getAccount.getUserId
+
+
```
Comment (mickel1138) [inline src/Pelagos/Bundle/AppBundle/Resources/views/Email/Account/UsernameRetrieval.email.twig]: one too many line breaks?

File: src/Pelagos/Bundle/AppBundle/Resources/views/Account/forgotUsername.html.twig [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+                <h2 class="element-invisible">Success message</h2>
+                The Gulf of Mexico Research Initiative Information and Data Cooperative (GRIIDC) has confirmed
+                that your email address is associated with a GRIIDC account. Your username has been emailed to  {{ emailId }}.
+                For any additional help please contact<a href="griidc@gomri.org">griidc@gomri.org</a>.
+            </div>
+        {% else %}
+            <div class="messages error">
+                <h2 class="element-invisible">Error message</h2>
+                No account is associated with this email address : {{ emailId }}
+                For any additional help please contact<a href="griidc@gomri.org">griidc@gomri.org</a>.
```
Comment (mickel1138) [inline src/Pelagos/Bundle/AppBundle/Resources/views/Account/forgotUsername.html.twig]: needs space after "contact"

File: src/Pelagos/Bundle/AppBundle/Resources/views/Account/forgotUsername.html.twig [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+            <div>
+                <h2 class="element-invisible">Success message</h2>
+                The Gulf of Mexico Research Initiative Information and Data Cooperative (GRIIDC) has confirmed
+                that your email address is associated with a GRIIDC account. Your username has been emailed to  {{ emailId }}.
+                For any additional help please contact<a href="griidc@gomri.org">griidc@gomri.org</a>.
+            </div>
+        {% else %}
+            <div class="messages error">
+                <h2 class="element-invisible">Error message</h2>
+                No account is associated with this email address : {{ emailId }}
```
Comment (mickel1138) [inline src/Pelagos/Bundle/AppBundle/Resources/views/Account/forgotUsername.html.twig]: Maybe remove a space before "address"

**Comment by fightingtexasaggie** [comment]
From: Williamson, Michael <Michael.Williamson@tamucc.edu> 
Sent: Thursday, November 8, 2018 10:53 AM
To: Cassin, Lionel <Lionel.Cassin@tamucc.edu>
Subject: forgot username type tools
 
We're putting together a GRIIDC tool that emails a user their username in the event they forgot it. Since we establish our user accounts against validated email addresses to begin with, the logic works for us. The interface simply asks for an email address, then emails the username to that address associated with the account. It currently also gives an "email address not found" if it isn't in the system, and this is where my question comes about. Can we do this, or do we need to make sure the system cannot be used to validate email addresses. (i.e. no error message or any different behavior for unknown email addresses)
 
Thanks,
Michael

RESPONSE FROM OIS:
The latter.  You don’t want “oracles” – i.e., something that you can query to gain actionable intelligence.  So, I’d prefer it say
 
“If that email address is found in our system, we will send the associated username to the found email address”
 
Tx
LC

**Comment by praneethpr** [comment]
@fightingtexasaggie Made the required changes! 👍

File: src/Pelagos/Bundle/AppBundle/Resources/views/Account/forgotUsername.html.twig [hunk 4]
```diff
@@ -17,19 +17,10 @@
...
-                For any additional help please contact <a href="griidc@gomri.org">griidc@gomri.org</a>.
-            </div>
-        {% else %}
-            <div class="messages error">
-                <h2 class="element-invisible">Error message</h2>
-                No account is associated with this email address: {{ emailId }}
-                For any additional help please contact <a href="griidc@gomri.org">griidc@gomri.org</a>.
-            </div>
-        {% endif %}
+        <div>
```
Comment (mickel1138) [inline src/Pelagos/Bundle/AppBundle/Resources/views/Account/forgotUsername.html.twig]: Would you like to add `class="messages status"` here?

**Pull request merged**